JavaScript/testing-sdk/src/index.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  runTestSuite,
+  BaseTestEvaluator,
+  Evaluation,
+} from '@autoblocks/client/testing';
+
+interface MyTestCase {
+  x: number;
+  y: number;
+}
+
+class MyTestEvaluator extends BaseTestEvaluator<MyTestCase, number> {
+  id = 'my-test-evaluator';
+
+  evaluateTestCase(testCase: MyTestCase, output: number): Evaluation {
+    return {
+      score: testCase.x + output,
+      threshold: {
+        gte: 0.3,
+      },
+    };
+  }
+}
+
+const fn = (testCase: MyTestCase) => {
+  return new Promise<number>((resolve) => {
+    setTimeout(() => {
+      resolve(testCase.y);
+    }, 1_000);
+  });
+};
+
+async function main() {
+  await runTestSuite<MyTestCase, number>({
+    id: 'my-test-suite',
+    testCases: [
+      { x: 0.1, y: 0.1 },
+      { x: 0.2, y: 0.1 },
+    ],
+    testCaseHash: 'x',
```

Comment:
TS doesn't have a dataclasses equivalent and making test cases classes would be pretty annoying, so for TS there is an arg on the main `runTestSuite` interface where you can either give us a property to hash or a function that returns a string